### PR TITLE
Clean pandas version check less than 2

### PR DESCRIPTION
**Pull request by mroeschke**
## Description
Optional cleanup to remove unnecessary version checks for pandas. Non blocking for the pandas branch release

## Checklist
- [ ] I am familiar with the [Contributing Guidelines](https://github.com/rapidsai/cudf/blob/HEAD/CONTRIBUTING.md).
- [ ] New or existing tests cover these changes.
- [ ] The documentation is up to date with these changes.

### CODE REVIEW

File: python/cudf/cudf/core/_compat.py [hunk 1]
```diff
@@ -4,13 +4,7 @@
 from packaging import version
 
 PANDAS_VERSION = version.parse(pd.__version__)
-PANDAS_GE_133 = PANDAS_VERSION >= version.parse("1.3.3")
-PANDAS_GE_134 = PANDAS_VERSION >= version.parse("1.3.4")
-PANDAS_LT_140 = PANDAS_VERSION < version.parse("1.4.0")
-PANDAS_GE_150 = PANDAS_VERSION >= version.parse("1.5.0")
-PANDAS_LT_153 = PANDAS_VERSION < version.parse("1.5.3")
 PANDAS_EQ_200 = PANDAS_VERSION == version.parse("2.0.0")
```
Comment (bdice) [inline python/cudf/cudf/core/_compat.py:7]: Have we determined the minimum 2.x version that we intend to support? We may choose to require 2.1.x or higher. In that case we can clean this up further, at a later time.

File: python/cudf/cudf/tests/test_csv.py [hunk 2]
```diff
@@ -372,11 +371,10 @@ def test_csv_reader_skiprows_skipfooter(tmpdir, pd_mixed_dataframe):
...
     assert len(out.columns) == len(df_out.columns)
     assert len(out) == len(df_out)
-    if PANDAS_GE_200:
-        # TODO: Remove typecast to `ns` after following
-        # issue is fixed:
-        # https://github.com/pandas-dev/pandas/issues/52449
-        out["2"] = out["2"].astype("datetime64[ns]")
+    # TODO: Remove typecast to `ns` after following
+    # issue is fixed:
+    # https://github.com/pandas-dev/pandas/issues/52449
```
Comment (bdice) [inline python/cudf/cudf/tests/test_csv.py:376]: This issue is marked as closed, so maybe we can remove this TODO? Or condition the cast on some 2.x.x version?

File: python/cudf/cudf/tests/test_dataframe.py [hunk 3]
```diff
@@ -6651,20 +6600,14 @@ def test_df_series_dataframe_astype_dtype_dict(copy):
     [
         ([1, 2, 3, 100, 112, 35464], ["a"]),
         (range(100), None),
-        pytest.param(
+        (
```
Comment (bdice) [inline python/cudf/cudf/tests/test_dataframe.py:6603]: Do we want to reformat this section to remove the extra newlines / etc.?

File: python/cudf/cudf/tests/test_datasets.py [hunk 4]
```diff
@@ -1,4 +1,4 @@
-# Copyright (c) 2019-2022, NVIDIA CORPORATION.
+# Copyright (c) 2019-2024, NVIDIA CORPORATION.
```
Comment (bdice) [inline python/cudf/cudf/tests/test_datasets.py:1]: Unnecessary copyright update?

**Review by bdice — COMMENTED** [review]
Flushing comments. I didn't quite get through the end of the PR. I got up to `test_parquet.py`.

File: python/cudf/cudf/core/_compat.py (hunk 1 above)
Comment (galipremsagar) [reply python/cudf/cudf/core/_compat.py:7]: @mroeschke @bdice I think we can comfortably support `2.0` as minimum version. Agree, like any release the first major release has a good number of issues that were fixed in later `2.x` versions. But imposing `2.x` itself is a breaking step change for users and I feel skipping `2.0` as the minimum version shouldn't be done at least in `24.04`. 

Coming to this PR. Can we either:

1. Hold off on this PR until we are 1-2 weeks near the burndown end for `24.04`? 

or


2. Only drop all the versions until `pre-1.5.3` and keep the version checks from `1.5.3` in `pandas_2.0_feature_branch` and then drop the remaining `1.5.3` version checks 1-2 weeks near the burndown end for `24.04`?


I feel 2 is more work, The reason I'm proposing 1 or 2 is we are yet to see how libraries and users react to `2.x` support in `24.04` dev cycle and these version checks are vital for any decision change to support both 1.5.x and 2.x simultaneously, I know that's not going to happen but I'm being cautious here.

Comment (galipremsagar) [reply python/cudf/cudf/core/_compat.py:7]: cc: @shwina incase you have any thoughts on the above.

Comment (shwina) [reply python/cudf/cudf/core/_compat.py:7]: I agree, we don't want to choose a specific minor version as the minimum supported version. But I don't think we should be cautious about support 1.x/2.x simultaneously. It's likely not going to be feasible to do that.

**Comment by mroeschke** [comment]
I'll break up this PR to remove these checks per minor 1.x version and not include any 2.0 version checks so closing

**Pull request closed without merging**